_sources/functions_in_cpp/mixed_up_code.rst

```diff
@@ -0,0 +1,344 @@
+   Put the necessary blocks of code in the correct order.
+   -----
+   double priceWithTax (double price, double percentTax) {
+   =====
+   int priceWithTax (double price, int percentTax) {  #distractor
+   =====
+   double priceWithTax (price, percentTax) {  #distractor
+   =====
+      return (1 + percentTax / 100) * price;
+   =====
+      return (1 + percentTax) * price;  #distractor
+   =====
+      return percentTax * price;  #distractor
+   =====
+   }
+
+.. parsonsprob:: functions_in_cpp_mixed_up_code_3
+   :numbered: left
+   :adaptive:
+   :noindent:
+   :practice: T
+
+   Most assignments and tests are graded as a percentage, but final
+   grades are letters. Let's write the code for the percentToLetter function. 
+   percentToLetter takes a percentage and returns the corresponding
+   letter grade. A 90 and above is an 'A', an 80 and above is a 'B', a 70 and above
+   is a 'C', and anything under a 70 is an 'F'.
+   Put the necessary blocks of code in the correct order.
+   -----
+   char percentToLetter (double percentage) {
+   =====
+   void percentToLetter (double percentage) {  #distractor
+   =====
+   void percentToLetter (int percentage) {  #distractor
+   =====
+   char percentToLetter (percentage) {  #distractor
+   =====
+      if (percentage >= 90) {
+   =====
+         return 'A';
+   =====
+         return A;  #paired
+   =====
+      }
+   =====
+      else if (percentage >= 80) {
+   =====
+         return 'B';
+   =====
+         return 'B'  #paired
+   =====
+      }
+   =====
+      else if (percentage >= 70) {
+   =====
+      else if (percentage > 70) {  #paired
+   =====
+         return 'C';
+   =====
+      }
+   =====
+      else {
+   =====
+         return 'F';
+   =====
+      }
+   =====
+   }
+
+.. parsonsprob:: functions_in_cpp_mixed_up_code_4
+   :numbered: left
+   :adaptive:
+   :noindent:
+   :practice: T
+
+   Let's write the code for the triangleArea function. triangleArea
+   takes two parameters, base and height. It returns the 
+   area of the triangle using the formula 1/2 * base * height.
+   Put the necessary blocks of code in the correct order.
+   -----
+   double triangleArea (double base, double height) {
+   =====
+   int triangleArea (double base, double height) {  #distractor
+   =====
+   void triangleArea (double base, double height) {  #distractor
+   =====
+   double triangleArea (base, height) {  #distractor
+   =====
+      double area;  #distractor
+   =====
+      return 0.5 * base * height;
+   =====
+      cout << 0.5 * base * height << endl;  #distractor
+   =====
+   }
+
+.. parsonsprob:: functions_in_cpp_mixed_up_code_5
+   :numbered: left
+   :adaptive:
+   :practice: T
+
+   Let's write the code for the cylinderVolume function. cylinderVolume
+   takes two parameters, radius and height. It returns the 
+   volume of the cylinder using the formula pi * radius * radius * height.
+   Put the necessary blocks of code in the correct order.
```

Comment:
The exercises before this treat indentation differently than the exercises after this - the exercises before this do the indentation in the blocks themselves, while the exercises after this make the students do the indentation. I think it would probably be better to be consistent throughout the page.